crates/napi/src/threadsafe_function.rs

```diff
@@ -404,10 +418,9 @@ unsafe extern "C" fn call_js_cb<T: 'static, V: NapiRaw, R, ES>(
       sys::napi_get_and_clear_last_exception(raw_env, &mut error_result),
       sys::Status::napi_ok
     );
-    assert_eq!(
-      sys::napi_fatal_exception(raw_env, error_result),
-      sys::Status::napi_ok
-    );
+    // When shutting down, napi_fatal_exception sometimes returns another exception
+    let stat = sys::napi_fatal_exception(raw_env, error_result);
+    assert!(stat == sys::Status::napi_ok || stat == sys::Status::napi_pending_exception);
```

Comment:
Occasionally, V8 seems to start its env cleanup in the middle of a call to `call_js_cb`, and the JS function call will return with a status of `napi_pending_exception`. When that happens, it seems to get "stuck" in the napi_pending_exception state, and attempting to raise an exception with `napi_fatal_exception()` will return `napi_pending_exception` instead of `napi_ok`. Adjusted the assertion to handle this case.